### PR TITLE
All data and MC GTs updated with GEM electronics map

### PR DESCRIPTION
**Pull request by arunhep**
This PR updates the data and mc global tags with the GEM electronics map.
offline data GTs now also contain latest 2017 rereco conditions.

# Summary of changes in Global Tags 
 
## RunI data 
 
   * **RunI HLT processing** : [100X_dataRun2_HLT_frozen_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_frozen_v2) as [100X_dataRun2_HLT_frozen_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_frozen_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_HLT_frozen_v2/100X_dataRun2_HLT_frozen_v1): 
      * update of GEM electronics map
      * change of tag for the SiPixel templates for 0T

   * **RunI Offline processing** : [100X_dataRun2_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_v3) as [100X_dataRun2_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_v3/100X_dataRun2_v1): 
      * update of GEM electronics map
      * update of 2017 rereco conditions

## RunII data 
 
   * **RunII HLTHI processing** : [100X_dataRun2_HLTHI_frozen_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLTHI_frozen_v2) as [100X_dataRun2_HLTHI_frozen_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLTHI_frozen_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_HLTHI_frozen_v2/100X_dataRun2_HLTHI_frozen_v1): 
      * update of GEM electronics map
      * change of tag for the SiPixel templates for 0T

   * **RunII HLT relval processing** : [100X_dataRun2_HLT_relval_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_relval_v2) as [100X_dataRun2_HLT_relval_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_relval_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_HLT_relval_v2/100X_dataRun2_HLT_relval_v1): 
      * update of GEM electronics map
      * change of tag for the SiPixel templates for 0T

   * **RunII Offline processing** : [100X_dataRun2_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_v3) as [100X_dataRun2_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_v3/100X_dataRun2_v1): 
      * update of GEM electronics map
      * update of 2017 rereco conditions

   * **RunII Offline relval processing** : [100X_dataRun2_relval_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_relval_v3) as [100X_dataRun2_relval_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_relval_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_relval_v3/100X_dataRun2_relval_v1): 
      * update of GEM electronics map
      * update of 2017 rereco conditions

   * **RunII Prompt processing** : [100X_dataRun2_PromptLike_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_PromptLike_v2) as [100X_dataRun2_PromptLike_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_PromptLike_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_PromptLike_v2/100X_dataRun2_PromptLike_v1): 
      * update of GEM electronics map
      * change of tag for the SiPixel templates for 0T

   * **RunI HLT processing** : [100X_dataRun2_HLT_frozen_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_frozen_v2) as [100X_dataRun2_HLT_frozen_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_dataRun2_HLT_frozen_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_dataRun2_HLT_frozen_v2/100X_dataRun2_HLT_frozen_v1): 
      * update of GEM electronics map
      * change of SiPixel templates 0T tag

## Upgrade 
 
   * **PhaseII realistic scenario** : [100X_upgrade2023_realistic_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2023_realistic_v2) as [100X_upgrade2023_realistic_v1](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2023_realistic_v1) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2023_realistic_v2/100X_upgrade2023_realistic_v1): 
      * update of GEM electronics map

   * **PhaseI 2018 cosmics scenario** : [100X_upgrade2018cosmics_realistic_deco_v10](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018cosmics_realistic_deco_v10) as [100X_upgrade2018cosmics_realistic_deco_v9](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018cosmics_realistic_deco_v9) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2018cosmics_realistic_deco_v10/100X_upgrade2018cosmics_realistic_deco_v9): 
      * update of GEM electronics map
      * update of ZeroMaterial SIM geometry

   * **PhaseI 2018 design scenario** : [100X_upgrade2018_design_IdealBS_v8](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018_design_IdealBS_v8) as [100X_upgrade2018_design_IdealBS_v7](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018_design_IdealBS_v7) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2018_design_IdealBS_v8/100X_upgrade2018_design_IdealBS_v7): 
      * update of GEM electronics map
      * update of ZeroMaterial SIM geometry

   * **PhaseI 2018 realistic scenario** : [100X_upgrade2018_realistic_v12](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018_realistic_v12) as [100X_upgrade2018_realistic_v11](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_upgrade2018_realistic_v11) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2018_realistic_v12/100X_upgrade2018_realistic_v11): 
      * update of GEM electronics map
      * update of ZeroMaterial SIM geometry

   * **PostLS2 design scenario** : [100X_postLS2_design_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_postLS2_design_v3) as [100X_postLS2_design_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_postLS2_design_v2) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_postLS2_design_v3/100X_postLS2_design_v2): 
      * update of Jet probability b tagging calibrations
      * update of GEM electronics map

   * **PostLS2 realistic scenario** : [100X_postLS2_realistic_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_postLS2_realistic_v3) as [100X_postLS2_realistic_v2](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_postLS2_realistic_v2) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_postLS2_realistic_v3/100X_postLS2_realistic_v2): 
      * update of Jet probability b tagging calibrations
      * update of GEM electronics map

## 2017_MC 
 
   * **PhaseI 2017 cosmics peak scenario** : [100X_mc2017cosmics_realistic_peak_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017cosmics_realistic_peak_v4) as [100X_mc2017cosmics_realistic_peak_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017cosmics_realistic_peak_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_mc2017cosmics_realistic_peak_v4/100X_mc2017cosmics_realistic_peak_v3): 
      * update of GEM electronics map

   * **PhaseI 2017 cosmics scenario** : [100X_mc2017cosmics_realistic_deco_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017cosmics_realistic_deco_v4) as [100X_mc2017cosmics_realistic_deco_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017cosmics_realistic_deco_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_mc2017cosmics_realistic_deco_v4/100X_mc2017cosmics_realistic_deco_v3): 
      * update of GEM electronics map

   * **PhaseI 2017 design scenario** : [100X_mc2017_design_IdealBS_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017_design_IdealBS_v4) as [100X_mc2017_design_IdealBS_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017_design_IdealBS_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_mc2017_design_IdealBS_v4/100X_mc2017_design_IdealBS_v3): 
      * update of GEM electronics map

   * **PhaseI 2017 realistic scenario** : [100X_mc2017_realistic_v4](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017_realistic_v4) as [100X_mc2017_realistic_v3](https://cms-conddb.cern.ch/cmsDbBrowser/list/Prod/gts/100X_mc2017_realistic_v3) with the following [changes](https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_mc2017_realistic_v4/100X_mc2017_realistic_v3): 
      * update of GEM electronics map

### CODE REVIEW

**Comment by arunhep** [comment]
please test

**Comment by arunhep** [comment]
+1

**Comment by arunhep** [comment]
as agreed in ORP today, we have updated all relevant MC GTs with the same GEM electronics map as data tags and switched the flag to True to enable the reading of electronics map from DB

**Comment by arunhep** [comment]
@cmsbuild please test

**Comment by arunhep** [comment]
+1

**Comment by jshlee** [comment]
thank you @arunhep
As far as I was aware, the current GT will only work for slice tests.
I did ask for a full geometry MC mapping DB, but I dont think this is it.

For now, I think it would be better to just change useDBEMap to true using Modifier_run2_GEM_2017_cff.

**Comment by arunhep** [comment]
@jshlee 

> As far as I was aware, the current GT will only work for slice tests.
> I did ask for a full geometry MC mapping DB, but I dont think this is it.
> 
> For now, I think it would be better to just change useDBEMap to true using Modifier_run2_GEM_2017_cff.

but how will that be different from changing the flag in gemUnpacker_cfi.py ?

**Comment by perrotta** [comment]
Using the modifier for run2_GEM_2017 allows it being set only for Run2_2017 and Riun2_2018, while it remains "false" for all other periods (namely Run3 and Phase2). As of the current implementation here the usage of DB mapping is also allowed in the gem unpacker for Run3 and Phase 2 workflows. Therefore, if the mapping is not present, or wrong, for whatever else that Run2, the flag change should be restricted by that modifier

@jshlee @arunhep  , since it does not make crashing the Run3 and Phase2 workflows in the matrix, I suppose that something is present in the GT also for those eras, although possibly the wrong one: could you please explain what should we expect there?

Indeed, looking at the jenkins outputs of this PR, we can see that impact parameters get changed in the 2019 wf (11624.0):
image
(and BTag as well, of course). It looks like being originated by the switch, as it did not show up in the first tests of this PR when getting the map from DB was not allowed yet. 

@jshlee : 
- How can the GEM enter in the computation of that IP? (Let me remind that in Run3 and Phase2 eras the GEM local reco is allowed)
- Why we do not see the same also in the Phase2 wfs?

**Comment by arunhep** [comment]
@perrotta 

> Indeed, looking at the jenkins outputs of this PR, we can see that impact parameters get changed in the 2019 wf (11624.0):

its not because of GEM but also because for this there is also an update of jet probability b tagging calibrations : https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_postLS2_realistic_v3/100X_postLS2_realistic_v2

but for phase-2 only update is GEM electronics map : 
https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/100X_upgrade2023_realistic_v2/100X_upgrade2023_realistic_v1

i should add the description of GT changes..

**Comment by perrotta** [comment]
@arunhep : thank you, that makes sense. In fact, I did not notice that you added another commit yesterday besides the one that sets ```useDBEMap = cms.bool(True)```

**Comment by perrotta** [comment]
Still, @jshlee please comment about the usage of the current DB map in Run3 and Phase2 wfs

**Comment by jshlee** [comment]
The current DB would only pack and unpack gem hits for the slice test chambers only.
For this PR, it would be best to use Modifier_run2_GEM_2017_cff to change the useDBEMap to true.

I will test this further and report back

**Comment by jshlee** [comment]
just realised that gemUnpacker_cfi.py is not needed or used.

process.muonGEMDigis is the one that needs to be changed in
https://github.com/cms-sw/cmssw/blob/master/Configuration/StandardSequences/python/RawToDigi_cff.py#L113

File: EventFilter/GEMRawToDigi/python/gemUnpacker_cfi.py [hunk 1]
```diff
@@ -3,5 +3,5 @@
 muonGEMDigis = cms.EDProducer("GEMRawToDigiModule",
     InputLabel = cms.InputTag("rawDataCollector"),
     UnpackStatusDigis = cms.bool(False),
-    useDBEMap = cms.bool(False),
+    useDBEMap = cms.bool(True),
```
Comment (slava77) [inline EventFilter/GEMRawToDigi/python/gemUnpacker_cfi.py]: is this change in the master already?

Comment (jshlee) [reply EventFilter/GEMRawToDigi/python/gemUnpacker_cfi.py]: no. 
and this file isnt used and should be removed.

**Comment by jshlee** [comment]
@arunhep - i will fix the gem issue with backport of #22619
can you remove the changes applied to EventFilter/GEMRawToDigi/python/gemUnpacker_cfi.py

**Comment by arunhep** [comment]
@jshlee 
> can you remove the changes applied to EventFilter/GEMRawToDigi/python/gemUnpacker_cfi.py

done

**Comment by perrotta** [comment]
... and as such reco signature is not needed any more
The reco part of the adaptation of the GEMs to Run2 is now addressed by #22619

**Comment by arunhep** [comment]
please test

**Comment by arunhep** [comment]
tests are triggered just for the sake of confirmation.

**Comment by arunhep** [comment]
+1

**Comment by davidlange6** [comment]
Your PR is unmergeable. Please have a look and possibly rebase it.

**Comment by arunhep** [comment]
@cmsbuild please test

**Comment by arunhep** [comment]
@fabiocos what are the plans for this PR? I was hoping it to be integrated in 10.0.5
thanks.

**Comment by fabiocos** [comment]
@arunhep it looked lated for 10_0_5, whose main target was different. I will merge it, although I hope that operations move soon after Easter to 10_1_X

**Comment by fabiocos** [comment]
+1

**Pull request merged**